grafast/dataplan-pg/src/index.ts

```diff
@@ -427,15 +427,12 @@ export { version } from "./version.js";
 
 declare global {
   namespace GraphileConfig {
-    interface PgServiceConfiguration<
-      TAdaptor extends
-        keyof GraphileConfig.PgDatabaseAdaptorOptions = keyof GraphileConfig.PgDatabaseAdaptorOptions,
-    > {
+    interface PgServiceConfiguration<TAdaptorOptions = PgAdaptorOptions> {
```

Comment:
We should not have this; defaulting to using the adaptor options for the 'pg' adaptor doesn't make sense for the majority of other adaptors, and would be unsafe to use across the codebase.

Removing this default causes cascading failures. I can't push to your branch so I've pushed up a commit here https://github.com/graphile/crystal/commit/8d57b63cea2e0750506402e0779c177dcdc0c64b to address the beginnings of these, but I can't figure out how to get it to pass with solid typings without resorting to `any`. Please feel free to pull my changes into your branch and continue iterating.